department-of-veterans-affairs/caseflow
# APPEALS-30239 - Create and Format ‘View History’ table

Resolves [APPEALS-30239](https://jira.devops.va.gov/browse/APPEALS-30239)

# Description
Frontend component for individual claim history table

## Acceptance Criteria
- [ ] Write components(s) to create table with data.
    - [ ] History is sorted from most recent to oldest activity.
    - [ ] Data is sorted in rows and aligned to columns with the following headers: Date and Time, User, Activity, Details.
    - [ ] Information can be sorted ascending/descending based on Date and Time, User or Activity.
    - [ ] Information can be filtered by User or Activity columns.
    - [ ] Up to 15 activities will display on each page and subsequent pages will be created accordingly.


## Testing Plan

1. Log in as VHAADMIN 
2. Navigate to VHA decision review queue and pick an in progress task
3. Click the view history button
4. A table should appear with dummy data
5. Verify the correct default sort, column names, etc from the AC


# Frontend
## User Facing Changes
<img width="1216" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/17840614/8a56106d-c302-4740-bc7b-500aaf4c3ff1">
<img width="1135" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/17840614/ff7cfb03-1262-4e38-a6e0-b6a4cd1ac5d2">


## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
